arraycontext/container/traversal.py

```diff
@@ -583,7 +583,10 @@ def thaw(ary: ArrayOrContainerT, actx: ArrayContext) -> ArrayOrContainerT:
 
 # {{{ flatten / unflatten
 
-def flatten(ary: ArrayOrContainerT, actx: ArrayContext) -> Any:
+def flatten(
+        ary: ArrayOrContainerT, actx: ArrayContext, *,
+        leaf_class: Optional[type] = None,
```

Comment:
Does `unflatten` just do the right thing when run on a with-leaf-class-flattened thing? If so, that might be a good thing to also explicitly document.

Reply:
No, `unflatten` doesn't really work with this "mode" of `flatten`. Not sure if it's worth implementing that, since it's not really used anywhere at the moment.

Reply:
Added a sentence to say that `unflatten` will not roundtrip if `leaf_class` is given in 5e87623.